test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -115,6 +115,123 @@ class LuaHttpFilterTest : public testing::Test {
         .WillOnce(testing::ReturnRef(metadata_));
   }
 
+  // Set asynchronous to false to make the filter waiting for `httpCall()`'s response. This is the
+  // default behavior. And set `sampled` to false to set request_options.sampled_ to false to skip a
+  // span to be sampled. See: AsyncRequestImpl constructor in async_client_impl.cc for more details.
+  struct TestBasicHttpCallOptions {
+    bool empty_{true};
+    bool asynchronous_{false};
+    bool sampled_{true};
+
+    // This wraps the asynchronous and sampled flags as a table. For example,
+    // {
+    //   ["asynchronous"]: false,
+    //   ["sampled"]: false
+    // }
+    bool as_table_{false};
+  };
+
+  void testBasicHttpCall(const TestBasicHttpCallOptions& call_options) {
+    // `last_arg` is the last argument of `httpCall()`. By default it is empty (not provided).
+    std::string last_arg = EMPTY_STRING;
+
+    // When setting `last_arg` with an empty string, the default behavior is to wait for
+    // `httpCall()` to finish and the span is always sampled.
+    if (!call_options.empty_) {
+      if (call_options.as_table_) {
+        last_arg = fmt::format("{{ [\"asynchronous\"] = {}, [\"sampled\"] = {} }}",
+                               call_options.asynchronous_, call_options.sampled_);
+      } else {
+        last_arg = call_options.asynchronous_ ? "true" : "false";
+      }
+    }
+
+    const std::string end = call_options.asynchronous_ ? R"EOF()
```

Comment:
Using multi line string here is a bit hard to read, maybe regular string works better?

Reply:
Done.